server/service/orbit.go

```diff
@@ -152,31 +152,76 @@ func (svc *Service) GetOrbitConfig(ctx context.Context) (flags json.RawMessage,
 	if host.TeamID != nil {
 		teamAgentOptions, err := svc.ds.TeamAgentOptions(ctx, *host.TeamID)
 		if err != nil {
-			return nil, nil, notifs, err
+			return fleet.OrbitConfig{Notifications: notifs}, err
 		}
 
+		var opts fleet.AgentOptions
 		if teamAgentOptions != nil && len(*teamAgentOptions) > 0 {
-			var opts fleet.AgentOptions
 			if err := json.Unmarshal(*teamAgentOptions, &opts); err != nil {
-				return nil, nil, notifs, err
+				return fleet.OrbitConfig{Notifications: notifs}, err
+			}
+		}
+
+		mdmConfig, err := svc.ds.TeamMDMConfig(ctx, *host.TeamID)
+		if err != nil {
+			return fleet.OrbitConfig{Notifications: notifs}, err
+		}
+
+		var nudgeConfig bytes.Buffer
+		if mdmConfig != nil &&
+			mdmConfig.MacOSUpdates.Deadline != "" &&
+			mdmConfig.MacOSUpdates.MinimumVersion != "" {
+			if err := nudgeConfigTemplate.Execute(&nudgeConfig, mdmConfig.MacOSUpdates); err != nil {
+				return fleet.OrbitConfig{Notifications: notifs}, err
 			}
-			return opts.CommandLineStartUpFlags, opts.Extensions, notifs, nil
 		}
+
+		return fleet.OrbitConfig{
+			Flags:         opts.CommandLineStartUpFlags,
+			Extensions:    opts.Extensions,
+			Notifications: notifs,
+			NudgeConfig:   nudgeConfig.Bytes(),
+		}, nil
 	}
 
 	// team ID is nil, get global flags and options
 	config, err := svc.ds.AppConfig(ctx)
 	if err != nil {
-		return nil, nil, notifs, err
+		return fleet.OrbitConfig{Notifications: notifs}, err
 	}
 	var opts fleet.AgentOptions
 	if config.AgentOptions != nil {
 		if err := json.Unmarshal(*config.AgentOptions, &opts); err != nil {
-			return nil, nil, notifs, err
+			return fleet.OrbitConfig{Notifications: notifs}, err
+		}
+	}
+
+	var nudgeConfig bytes.Buffer
+	if config.MDM.MacOSUpdates.Deadline != "" &&
+		config.MDM.MacOSUpdates.MinimumVersion != "" {
+		if err := nudgeConfigTemplate.Execute(&nudgeConfig, config.MDM.MacOSUpdates); err != nil {
+			return fleet.OrbitConfig{Notifications: notifs}, err
 		}
 	}
-	return opts.CommandLineStartUpFlags, opts.Extensions, notifs, nil
+
+	return fleet.OrbitConfig{
+		Flags:         opts.CommandLineStartUpFlags,
+		Extensions:    opts.Extensions,
+		Notifications: notifs,
+		NudgeConfig:   nudgeConfig.Bytes(),
+	}, nil
+}
+
+var nudgeConfigTemplate = template.Must(template.New("").Option("missingkey=error").Parse(`
+{
+  "osVersionRequirements": [
+    {
+      "requiredInstallationDate": "{{ .Deadline }}",
+      "requiredMinimumOSVersion": "{{ .MinimumVersion }}"
+    }
+  ]
```

Comment:
note that this won't be the final form of the config, there are a couple of things missing:

1. Adjusting the UI and copy to look like the spec
2. Set the correct timezone for the deadline

However, this is good enough to unblock @gillespi314 , and I will be following up soon in another PR.